homeassistant/components/switcher_kis/__init__.py

```diff
@@ -58,23 +72,52 @@
         """On homeassistant stop, gracefully stop the bridge if running."""
         await v2bridge.stop()
 
-    hass.async_add_job(hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STOP, async_stop_bridge))
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_bridge)
 
     try:
         device_data = await wait_for(
             v2bridge.queue.get(), timeout=5.0, loop=hass.loop)
     except (Asyncio_TimeoutError, RuntimeError):
-        _LOGGER.exception("failed to get response from device")
+        _LOGGER.exception("Failed to get response from device")
         await v2bridge.stop()
         return False
 
     hass.data[DOMAIN] = {
         DATA_DEVICE: device_data
     }
 
+    async def async_switch_platform_discoverd(
+            platform: str, discovery_info: Optional[Dict]) -> None:
+        """Use for registering services after switch platform is discoverd."""
+        async def async_set_auto_off_service(service: ServiceCallType) -> None:
```

Comment:
We should check that the correct platform was discovered. The listener will fire for all switch platforms.

Reply:
@MartinHjelmare 
Thank you very much!

Do you perhaps have any preferred way of accomplishing that other then planting a value in the discovery info dict?

Reply:
We have access to the `platform` argument.
```py
if platform != DOMAIN:
    return
```

Reply:
@MartinHjelmare 
Hmm... I was sure that will return `switch` in this case...
Thanks, I'll push the resolved code shortly.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/23477/commits/5d6e266b8483a18dd9cf8205177f6a469d998765).